app/grandchallenge/components/models.py

```diff
@@ -227,15 +243,34 @@ class ComponentInterfaceValue(models.Model):
     interface = models.ForeignKey(
         to=ComponentInterface, on_delete=models.CASCADE
     )
-    value = models.JSONField(null=True, default=None)
+    value = models.JSONField(null=True, blank=True, default=None)
     file = models.FileField(
-        upload_to=component_interface_value_path, storage=protected_s3_storage
+        null=True,
+        blank=True,
+        upload_to=component_interface_value_path,
+        storage=protected_s3_storage,
+        validators=[
+            ExtensionValidator(allowed_extensions=(".json", ".zip", ".csv")),
+            MimeTypeValidator(
+                allowed_types=(
+                    "application/json",
+                    "application/zip",
+                    "text/plain",
+                )
+            ),
+        ],
+    )
+    image = models.ForeignKey(
+        to=Image, null=True, blank=True, on_delete=models.CASCADE
     )
-    image = models.ForeignKey(to=Image, null=True, on_delete=models.CASCADE)
 
     def __str__(self):
         return f"Component Interface Value {self.pk} for {self.interface}"
 
+    def save(self, *args, **kwargs):
+        self.full_clean()
```

Comment:
I'm not in favour of adding `full_clean` to save as it means that any callers of `save` now need to handle a `ValidationError`, and I'm not confident that this is handled everywhere within django itself. Could you move this to where it is needed?

Reply:
I think this is to handle the `CIV.create()`, right? Without doing `full_clean` in `save`, `CIV.create(...)` would become:

```
CIV(...)
CIV.full_clean()  # This line would then go in a try: ... except ValidationError: ... block for handling
CIV.save()
```